pkg/planner/core/point_get_plan.go

```diff
@@ -1909,9 +1909,22 @@ func getPartitionDef(ctx sessionctx.Context, tbl *model.TableInfo, pairs []nameV
 	case model.PartitionTypeKey:
 		// The key partition table supports FastPlan when it contains only one partition column
 		if len(pi.Columns) == 1 {
+			// We need to change the partition column index!
+			partCols, _ := partitionExpr.GetPartColumnsForKeyPartition(schema.Columns)
+			if len(partCols) != 1 {
+				return nil, 0, 0, false
+			}
+			pe := &tables.ForKeyPruning{KeyPartCols: partCols}
+			colName := ""
```

Comment:
How about use `pe.KeyPartCols[0].OrigName` instead the for loop? I think they are same (not 100% sure about this).

Reply:
Thanks for the suggestion, I updated the code.